final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,150 @@
+    is xml
+    :param url: url provided by user
+    :param verbose: True or False
+    :return: True in case of xml,
+    False and closing program in all other cases
+    """
+    if verbose:
+        print('Wait a second, we will check if there is any RSS in URL you have given')
+        print('-' * 50)
+    response = get(url)
+    r = response.headers['Content-Type']  # We will check only headers
+    if 'xml' not in r:
+        print('Something went wrong. \n'
+              'Looks like URL you\'ve entered doesn\'t content any RSS feed')
+        return False
+    else:
+        return True
+
+
+def get_feed(url: str, verbose: bool):
+    """
+    Caching feed by creating a feedparser instance.
+    Returns non-formatted feed
+    :param url: url provided by user
+    :param verbose: True of False
+    :return: non-formatted feed
+    """
+    if verbose:
+        print('Establishing connection')
+        print('-' * 50)
+    feed = feedparser.parse(url)
+    if feed is None:
+        print('STOP')
+    if verbose:
+        print('Connection established')
+        print('-' * 50)
+    return feed
+
+
+def format_feed(feed, verbose: bool, limit: int):
+    """
+    This function recieves feed and format all entries
+    by using them as to_text() arguments to remove
+    all HTML markup
+    :param feed: Cached feed returned by get_feed()
+    :param verbose: True or False
+    :param limit: Standart limit is 1 article
+    :return: title, entry_title(Article title), description(Text of article), date
+    """
+    global title, entry_title, description, date
+    if verbose:
+        print('Formatting your feed')
+        print('-' * 50)
+    try:
+        title = to_text(feed.feed.title)
+        entry_title = to_text(feed.entries[limit].title)
+        description = to_text(feed.entries[limit].description)
+        date = to_text(feed.entries[limit].published)
+    except AttributeError as error: # in case of AttributeError exception
+        with open('log' + str(datetime.now()) + '.txt', 'w') as log:  # we will create a .txt log-file
+            log.write('[URL]: ' + str(url) + '\n')  # providing info about feed that raised an exception
+            log.write('[ERROR]: ' + str(error) + '\n')  # and text of error
+        print('An error has occured \n'
+              'Log file was created in program folder \n'
+              'To help us debug a program, please send this file to h4j0rx@gmail.com')
+    return title, entry_title, description, date
+
+
+def main(title, entry_title, description, date):
+    """
+    This function just prints all parameters given by format_feed()
+    :param title: Title of whole feed returned by format_feed()
+    :param entry_title: Title of one article returned by format_feed()
+    :param description: Description of one article returned by format_feed()
+    :param date: Publishing date of one article returned by format_feed()
+    :return: None
+    """
+    print('[URL]: ' + args.url)
```

Comment:
It is better to make URL for each piece of news actual URL to the webpage containing this piece of news.